tsconfig.json

```diff
@@ -0,0 +1,51 @@
+{
+  "compilerOptions": {
+    // Module Resolution
+    "baseUrl": "./",
+    "moduleResolution": "node",
+    "module": "esnext",
+    "paths": {
+      "@shared/*": ["app/shared/*"]
+    },
+    "resolveJsonModule": true,
+    "esModuleInterop": true,
+    "allowSyntheticDefaultImports": true,
+
+    // Compilation Target & Features
+    "target": "ES2020",
+    "lib": ["dom", "dom.iterable", "esnext"],
+    "jsx": "preserve",
+
+    // Type Checking & Safety
+    "strict": true,
+    "strictNullChecks": true,
+    "forceConsistentCasingInFileNames": true,
+    "skipLibCheck": true,
+    "isolatedModules": true,
+    "noImplicitAny": false,
+    "strictBindCallApply": false,
+    "noFallthroughCasesInSwitch": false,
+
+    // Build Options
+    "allowJs": true,
+    "noEmit": true,
+    "incremental": true,
+
+    // Next.js Specific
+    "plugins": [
+      {
+        "name": "next"
+      }
+    ]
+  },
+  // File Inclusion/Exclusion
+  "include": [
+    "next-env.d.ts",
+    "**/*.ts",
+    "**/*.tsx",
+    "**/*.js",
+    "**/*.jsx",
+    ".next/types/**/*.ts"
+  ],
+  "exclude": ["node_modules"]
+}
```

Comment:
Yeah.. lets go ahead and tackle .js->.ts transitioning after we get the api-rewrite effort dev-complete.